react-ui/src/Components/profile/Profile.tsx

```diff
@@ -0,0 +1,183 @@
+        open={currentPasswordModalOpen} 
+        okText='Submit' 
+        okButtonProps={{form:'confirmCurrentPasswordForm', htmlType:'submit'}}
+        onCancel={() => {setCurrentPasswordModalOpen(false)}}
+        closable={false}
+        maskClosable={false}
+        destroyOnClose={true}
+      >
+        <Form id='confirmCurrentPasswordForm' onFinish={async () => {
+          try {
+            await post('api/updateUser', newUserInformation);
+            message.success('User information updated successfully!');
+            setCurrentPasswordModalOpen(false);
+          } catch {
+            message.error('Unable to update user information. Please try again later.');
+          }
+        }}>
+          <Form.Item
+            style={{paddingTop: '25px'}}
+            name='currentPassword'
+            validateTrigger='onBlur'
+            rules={[
+              { required: true, message: 'Please enter a password.' },
+              () => ({
+                async validator(_, currentPassword) {
+                  setCurrentPasswordValidating(true);
+                  try {
+                    if (await currentPasswordCorrect(userInfo?.username, currentPassword)) {
+                      setCurrentPasswordValidating(false);
+                      return Promise.resolve();
+                    }
+                    setCurrentPasswordValidating(false);
+                    return Promise.reject(new Error('Incorrect Password.'));
+                  } catch {
+                    return Promise.reject(new Error('Unable to validate password.'));
+                  }
+                },
+              })
+            ]}
+          >
+            <Input.Password
+              prefix={<LockOutlined />}
+              placeholder="Password"
+            />
+          </Form.Item>
+        </Form>
+      </Modal>
+      <Row style={{paddingTop:'1vh'}}>
+        <Divider/>       
+        <Col span={4}></Col>
+        <Col style={{paddingLeft:'5%', paddingRight:'5%'}} span={10}>
+          {loading === true ? 
+            (
+              <Space direction='vertical' size={'large'} style={{width: "100%"}}>
+                <Skeleton.Input active={true} size={'default'} block={true}/>
+                <Skeleton.Input active={true} size={'default'} block={true}/>
+                <Skeleton.Input active={true} size={'default'} block={true}/>
+                <Skeleton.Input active={true} size={'default'} block={true}/>
+                <Skeleton.Input active={true} size={'default'} block={true}/>
+                <Skeleton.Input active={true} size={'default'} block={true}/>
+                <Skeleton.Input active={true} size={'default'} block={true}/>
+                <Skeleton.Input active={true} size={'default'} block={true}/>
+              </Space>
+            ) : (
+              <Form 
+                id='ProfileForm' 
+                onFinish={(values) => {
+                  console.log(values)
```

Comment:
We’ll have to remember to take this out on our next project. Not enough to stop this pull request from getting merged.